q2_demux/plugin_setup.py

```diff
@@ -160,53 +160,30 @@
         citations['hamady2009']]
 )
 
+T = TypeMatch([SequencesWithQuality, PairedEndSequencesWithQuality])
 plugin.methods.register_function(
-    function=q2_demux.partition_samples_single,
-    inputs={'demux': SampleData[SequencesWithQuality]},
+    function=q2_demux.partition_samples,
+    inputs={'demux': SampleData[T]},
     parameters={'num_partitions': Int % Range(1, None)},
     outputs=[
-        ('partitioned_demux', Collection[SampleData[SequencesWithQuality]]),
+        ('partitioned_demux', Collection[SampleData[T]]),
     ],
     input_descriptions={
         'demux': 'The demultiplexed sequences to partition.'
     },
     parameter_descriptions={
-        'num_partitions': 'The number of partitions to split the demux into.'
-                          ' Defaults to partioning into individual samples.'
+        'num_partitions': 'The number of partitions to split the '
+                          'demultiplexed sequences into. Defaults to '
+                          'partioning into individual samples.'
     },
     output_descriptions={
-        'partitioned_demux': 'The demultiplexed sequences partitioned into '
-        'samples.'
+        'partitioned_demux': 'The partitioned demultiplexed sequences.'
     },
     name='Split demultiplexed sequence data into partitions.',
     description=('Partition demultiplexed single end sequences into '
```

Comment:
Since you've combined single and paired into one method, I'd update the description to 'single or paired end sequences' just so it's clear that this supports both types.

Reply:
Had to undo that the unioning of return view types does not actually work. Works for inputs but not outputs